packages/core/test/new-api/call.ts

```diff
@@ -186,6 +192,228 @@ describe("call", () => {
     assert.equal(callFuture.from, "0x2");
   });
 
+  it("Should be able to pass an AccountRuntimeValue as from option", () => {
+    const moduleWithDependentContractsDefinition = defineModule(
+      "Module1",
+      (m) => {
+        const example = m.contract("Example");
+
+        m.call(example, "test", [], { from: m.getAccount(1) });
+
+        return { example };
+      }
+    );
+
+    const constructor = new ModuleConstructor();
+    const moduleWithDependentContracts = constructor.construct(
+      moduleWithDependentContractsDefinition
+    );
+
+    assert.isDefined(moduleWithDependentContracts);
+
+    const callFuture = [...moduleWithDependentContracts.futures].find(
+      ({ id }) => id === "Module1:Example#test"
+    );
+
+    if (!(callFuture instanceof NamedContractCallFutureImplementation)) {
+      assert.fail("Not a named contract deployment");
+    }
+
+    assertInstanceOf(callFuture.from, AccountRuntimeValueImplementation);
+    assert.equal(callFuture.from.accountIndex, 1);
+  });
+
+  describe("Arguments", () => {
```

Comment:
I added test of `ArgumentType`s for every future that takes either constructorArguments or arguments.